app/lib/MyAccountsStore.js

```diff
@@ -86,6 +87,59 @@ export default class MyAccountsStore {
       .then(() => this.updateConnector(connector))
   }
 
+  fetchOrInstallConnector (slug) {
+    const connectorSpec = this.connectors.find(conn => conn.slug === slug)
+    this.fetchConnector(connectorSpec)
+    .then(connector => {
+      if (connector === null) {
+        return this.installConnector(connectorSpec)
+      } else {
+        return connector
+      }
+    })
+  }
+
+  // Fetch stack data for the given connector
+  fetchConnector (connector) {
+    return cozy.client.data.defineIndex('io.cozy.konnectors', ['slug'])
+      .then(index => cozy.client.data.query(index, {selector: {slug: connector.slug}}))
+      .then(list => {
+        if (list.length) {
+          return list[0]
+        } else {
+          return null
+        }
+      })
+      .then(connector => {
+        if (connector) this.updateConnector(connector)
+        return connector
+      })
+  }
+
+  // Trigger the installation of the given connector
+  // and check that the installation is terminated
+  installConnector (connector) {
+    return this.fetch('POST', `/konnectors/${connector.slug}?Source=${encodeURIComponent(connector.repo)}`)
+      .then(() => {
+        // monitor the status of the connector
+        return new Promise((resolve, reject) => {
+          const idInterval = window.setInterval(() => {
```

Comment:
We should avoid to rely on `window` object. It could make a unit or function test to fail. A simple call to `setInterval` should do the job here.